### PR TITLE
chore(deps): update to Davinci@17

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-2831]

### Description

Update to Davinci@17
- use cypress@10

First commit just updates config and necessary stuff to run cypress@10

Second commit is refactor of all cypress test files. As suggested in Cypress documentation all needed providers for component tests should be added on one place in commands, where we create custom `cy.mount`

```diff
- import { mount } from '@cypress/react'
- import { TestingPicasso } from '@toptal/picasso/test-utils'

- mount(<TestingPicasso><Component /></TestingPicasso>
+cy.mount(<Component />)
```

### How to test

- Happo screenshots should not change
- CI should be green
- run locally to see the new cypress@10 design `yarn test:integration:open`

### Development checks

- Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- Annotate all `props` in component with documentation
- Create `examples` for component
- Ensure that deployed demo has expected results and good examples
- Ensure that tests pass by running `yarn test`
- Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- Covered with tests

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2831]: https://toptal-core.atlassian.net/browse/FX-2831?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -121,7 +119,9 @@
     "typescript": "^4.6.3",
     "webpack-dev-server": "^4.7.4",
     "yargs": "^13.2.4",
-    "yarn-deduplicate": "^3.1.0"
+    "yarn-deduplicate": "^3.1.0",
+    "url-loader": "^4.1.1"
```
Comment (denieler) [inline package.json:123]: why do we need that?

File: cypress/component/Accordion.spec.tsx [hunk 2]
```diff
@@ -5,10 +5,8 @@ import {
   Typography,
   Button,
   Container,
-  Check16
+  Check16,
 } from '@toptal/picasso'
-import { mount } from '@cypress/react'
-import { TestingPicasso } from '@toptal/picasso/test-utils'
```
Comment (TomasSlama) [inline cypress/component/Accordion.spec.tsx:11]: on this file I just showcase how we can improve our files

File: cypress/component/Accordion.spec.tsx [hunk 3]
```diff
@@ -154,7 +136,7 @@ describe('Accordion with custom summary', () => {
   })
 
   it('interacts with accordion content', () => {
-    mount(<AccordionCustomSummary />)
+    cy.mount(<AccordionCustomSummary />)
```
Comment (separatio) [inline cypress/component/Accordion.spec.tsx:139]: I definitely agree with using `cy.mount` vs `mount`. :+1:

File: package.json [hunk 4]
```diff
@@ -24,9 +24,9 @@
     "symlink": "cross-env ./bin/symlink --link",
     "symlink:off": "cross-env ./bin/symlink --unlink",
     "test": "yarn test:unit && yarn test:integration",
-    "test:integration": "yarn build:package && yarn test:setup cypress run-ct",
-    "test:integration:ci": "yarn happo-e2e -- -- yarn cypress run-ct",
-    "test:integration:open": "yarn build:package && yarn test:setup cypress open-ct",
+    "test:integration": "yarn build:package && yarn test:setup cypress run --component",
+    "test:integration:ci": "yarn happo-e2e -- -- yarn cypress run --component",
+    "test:integration:open": "yarn build:package && yarn test:setup cypress open --browser=electron --component",
```
Comment (separatio) [inline package.json:29]: Why Electron? Why not Chrome?

Comment (TomasSlama) [reply package.json:29]: It is up to us to change it if we want. But by default cypress is downloaded with electron and we used it till now.

File: cypress/support/commands.jsx [hunk 5]
```diff
@@ -24,3 +29,9 @@ Cypress.Commands.add('getByTestId', (testId, options) => {
 Cypress.Commands.add('getByRole', (role, options) => {
   return cy.get(`[role=${role}]`, options)
 })
+
+Cypress.Commands.add('mount', (component, options, props = {}) => {
+  // Wrap any parent components needed
+  // ie: return mount(<MyProvider>{component}</MyProvider>, options)
+  return mount(<TestingPicasso {...props}>{component}</TestingPicasso>, options)
+})
```
[the remark this answers is not in the document]
Comment (TomasSlama) [reply cypress/support/commands.jsx:37]: Cypress has this as best practice and it actually cleans the test files.

File: cypress.config.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
+const path = require('path')
+const happoTask = require('happo-cypress/task')
+/* eslint-disable import/no-extraneous-dependencies */
+const { defineConfig } = require('cypress')
```
Comment (denieler) [inline cypress.config.ts:4]: if I'm not mistaken in `cypress.config.ts` we can use `import { defineConfig } from 'cypress'` syntax 👍

File: package.json [hunk 7]
```diff
@@ -123,10 +120,11 @@
     ]
   },
   "resolutions": {
+    "cypress": "^10.0.2",
```
Comment (denieler) [inline package.json:123]: I don't think we even need a resolution

File: cypress/component/ApplicationUpdateNotification.spec.tsx [hunk 8]
```diff
@@ -32,22 +30,14 @@ const ApplicationUpdateNotificationExample = () => {
 
 describe('ApplicationUpdateNotification', () => {
   it('renders notification when click on trigger', () => {
-    mount(
-      <TestingPicasso>
```
Comment (denieler) [inline cypress/component/ApplicationUpdateNotification.spec.tsx:36]: now I get why we need a special `mount` command 😃 👍 🎉 looks amazing!

File: cypress/support/commands.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (TomasSlama) [reply cypress/support/commands.jsx:37]: @denieler ☝️

File: package.json (hunk 7 above)
Comment (TomasSlama) [reply package.json:123]: @denieler sadly yes, happo has version 9 in dependencies

File: package.json [hunk 9]
```diff
@@ -123,10 +120,11 @@
...
     "lodash": "^4.17.21",
     "**/marksy/marked": "^0.6.0",
     "@storybook/react": "^6.4.21",
-    "webpack": "^4.0.0",
+    "webpack": "^4",
```
[the remark this answers is not in the document]
Comment (TomasSlama) [reply package.json:127]: 👍, I was just checking why we need it. And forgot to undo the changes.

**Review by OndrejTuma — APPROVED** [review]
I have to join on praising the new `cy.mount` command 🎉 
Really clear things up and makes creating new cypress component test easier 👍

**Review by ozgurkececioglu — APPROVED** [review]
💪

File: package.json (hunk 7 above)
Comment (denieler) [reply package.json:123]: I was checking yesterday and it was working for me 🤷‍♂️ can I ask you please to double-check? 🙇‍♂️

**Comment by separatio** [comment]
PR looks good. Nothing to add, nothing to subtract. :+1:

**Review by separatio — APPROVED** [review]

**Pull request merged**